packages/core/src/render3/instructions/shared.ts

```diff
@@ -1265,13 +1290,14 @@ export function invokeHostBindingsInCreationMode(def: DirectiveDef<any>, directi
  * If a component is matched (at most one), it is returned in first position in the array.
  */
 function findDirectiveDefMatches(
-    tView: TView, lView: LView,
-    tNode: TElementNode|TContainerNode|TElementContainerNode): DirectiveDef<unknown>[]|null {
+    tView: TView, lView: LView, tNode: TElementNode|TContainerNode|TElementContainerNode):
+    [matches: DirectiveDef<unknown>[], definitionMap: HostDirectiveDefinitionMap|null]|null {
   ngDevMode && assertFirstCreatePass(tView);
   ngDevMode && assertTNodeType(tNode, TNodeType.AnyRNode | TNodeType.AnyContainer);
 
   const registry = tView.directiveRegistry;
   let matches: DirectiveDef<unknown>[]|null = null;
+  let definitionMap: HostDirectiveDefinitionMap|null = null;
```

Comment:
I'd propose renaming the `let` to be more specific:
```suggestion
  let hostDirectiveDefs: HostDirectiveDefs|null = null;
```